src/FileStrategy.ts

```diff
@@ -0,0 +1,90 @@
+import GitDB from './gitdb';
+import fse from 'fs-extra';
+import { Filter, SetCallback, DBRecord } from './collection';
+
+interface Update {
+  updated: boolean;
+  filePaths: string[];
+}
+
+interface Delete {
+  deleted: boolean;
+  filePaths: string[];
+}
+
+export class FileStrategy<T extends DBRecord> {
+  private db: GitDB;
+  private name: string;
+
+  constructor(gitDb: GitDB, name: string) {
```

Comment:
not really sure FileStrategy should depend on gitDb and name. 
Maybe `absoluteDirPath` param alone can replace these dependencies?